<?xml version="1.0"?>
<launch>
    <arg name="obs_abs_width"   value="0.72"/>
    <arg name="obs_abs_depth"   value="0.52"/>
    <arg name="map_abs_height"  value="4.0"/>
    <arg name="map_abs_width"   value="5.0"/>
    <arg name="InflateLength"   value="0.5"/> <!-- 0.1 -->

    <!-- astar node -->
    <node pkg="astar" type="astar_dj" name="astar_dj" output="screen">
        <param name="Euclidean"         value="true"/>
        <param name="InflateLength"     value="$(arg InflateLength)"/> 
        <param name="rate"              value="10"/>
        <param name="map_abs_height"    value="$(arg map_abs_height)"/>
        <param name="map_abs_width"     value="$(arg map_abs_width)"/>
        <param name="obs_abs_width"     value="$(arg obs_abs_width)"/>
        <param name="obs_abs_depth"     value="$(arg obs_abs_depth)"/>
        <param name="map_grid_length"   value="0.05"/>
    </node>

    <!-- visualize obstacle -->
    <node pkg="astar" type="obs_visualize" name="obs_visualize" output="screen">
        <param name="obs_abs_width"     value="$(arg obs_abs_width)"/>
        <param name="obs_abs_depth"     value="$(arg obs_abs_depth)"/>
        <param name="obs_abs_height"    value="1.0"/>
        <param name="map_abs_height"    value="$(arg map_abs_height)"/>
        <param name="map_abs_width"     value="$(arg map_abs_width)"/>
        <param name="InflateLength"     value="$(arg InflateLength)"/> 
    </node>

    <!-- publisher test for simulation -->
    <!-- <node pkg="astar" type="publish_simul" name="publish_simul" output="screen">
        <param name="yaw_angle"         value="30.0"/>
    </node> -->

    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find astar)/rviz/astar_dj.rviz"/>
</launch>